apps/prairielearn/elements/pl-string-input/pl-string-input.mustache

```diff
@@ -1,66 +1,124 @@
+<script>
+    $(function(){
```

Comment:
For my own understanding, why does this need to be wrapped up in a jQuery block?

Reply:
This needs to execute only after the document is loaded, since at the point where this script runs some DOM elements are not yet loaded. In client asset code we typically use "onDocumentLoaded", but that is not currently available in element code. Calling a jquery block with a function [executes it on load](https://learn.jquery.com/using-jquery-core/document-ready/).

Reply:
Why is `document.onload` not available in element code?

Reply:
From https://api.jquery.com/ready/:

> Most browsers [provide similar functionality](https://caniuse.com/#search=DOMContentLoaded) in the form of a DOMContentLoaded event. However, jQuery's .ready() method differs in an important and useful way: If the DOM becomes ready and the browser fires DOMContentLoaded before the code calls .ready( handler ), the function handler will still be executed. In contrast, a DOMContentLoaded event listener added after the event fires is never executed.

Reply:
Also, the onload method (contrary to DOMContentLoaded) waits until images and stylesheets are loaded, which is not necessary here.

Reply:
Resolving this conversation to clean up the discussion. @reteps if you still have concerns, you can reopen this thread.